components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -3699,6 +3771,37 @@ private void addIdPRoles(Connection conn, int idPId, int tenantId, String[] idpR
         }
     }
 
+    /**
+     * Add Identity Provider groups.
+     *
+     * @param conn          Database connection.
+     * @param idPId         Identity Provider ID.
+     * @param tenantId      Tenant ID of the Identity Provider.
+     * @param idpGroupNames List of Identity Provider groups.
+     * @throws SQLException Database exception when adding Identity Provider groups.
+     */
+    private void addIdPGroups(Connection conn, int idPId, int tenantId, String[] idpGroupNames)
+            throws SQLException {
+
+        if (idpGroupNames == null || idpGroupNames.length == 0) {
+            return;
+        }
+
+        try (PreparedStatement addIdPGroupsPrepStmt = conn.prepareStatement(
+                IdPManagementConstants.SQLQueries.ADD_IDP_GROUPS_SQL)) {
+            for (String idpGroup : idpGroupNames) {
+                addIdPGroupsPrepStmt.setInt(1, idPId);
+                addIdPGroupsPrepStmt.setInt(2, tenantId);
+                addIdPGroupsPrepStmt.setString(3, idpGroup);
+                addIdPGroupsPrepStmt.addBatch();
+                addIdPGroupsPrepStmt.clearParameters();
+            }
+
```

Comment:
Remove extra new line

Reply:
Addressed in 6105689ade57a923cad250a243246853d6167b51